nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -1360,15 +1705,20 @@ def exec(self):
 
         return IntersectsQuery(config1_without_ingress, self.config2, self.output_config).exec(True)
 
-    def compute_query_output(self, query_answer, cmd_line_flag=False):
-        res = not query_answer.numerical_result if cmd_line_flag else query_answer.bool_result
-        query_output = query_answer.output_result + '\n'
-        if query_answer.bool_result and not query_answer.query_not_executed:
-            query_output += f'{self.config2.name} does not forbid connections specified in {self.config1.name}:' \
-                            f'{query_answer.output_explanation}'
-        elif query_answer.numerical_result == 1 and not query_answer.query_not_executed:
-            query_output += f'{self.config2.name} forbids connections specified in {self.config1.name}'
-        return res, query_output, query_answer.query_not_executed
+    def execute_and_write_output_in_required_format(self, cmd_line_flag=False):
+        query_answer = self.exec()
+        query_result = not query_answer.numerical_result if cmd_line_flag else query_answer.bool_result
+        if query_answer.numerical_result == 1:
+            query_answer.output_result += f'\n{self.output_config.secondConfigName} forbids connections specified in ' \
+                                          f'{self.output_config.configName}'
+        if query_answer.output_explanation:
+            query_answer.output_result = f'{self.output_config.secondConfigName} does not forbid connections specified ' \
+                                         f'in {self.output_config.configName}'
+        explanation_header = f'Both {self.output_config.configName} and ' \
+                             f'{self.output_config.secondConfigName} allow the following connection(s)'
+        return query_result,\
+            self.write_query_output(query_answer, explanation_header, [explanation_header], int(query_result)), \
+            query_answer.query_not_executed
```

Comment:
Can try to simplify this as follows? 
extend `exec `function to consider the cmd line flag:

```
    def exec(self, cmd_line_flag=False):
        if not self.config1:
            return QueryAnswer(False, 'There are no NetworkPolicies in the given forbids config. '
                                      'No traffic is specified as forbidden.', query_not_executed=True)
        if self.config1.policies_container.layers.does_contain_single_layer(NetworkLayerName.Ingress):
            return QueryAnswer(bool_result=False,
                               output_result='Forbidden traffic cannot be specified using Ingress resources only',
                               query_not_executed=True)

        config1_without_ingress = self.clone_without_ingress(self.config1)

        res = IntersectsQuery(config1_without_ingress, self.config2, self.output_config).exec(True)
        res.bool_result = not res.numerical_result if cmd_line_flag else res.bool_result
        if res.numerical_result == 1:
            res.output_result += f'\n{self.output_config.secondConfigName} forbids connections specified in ' \
                                          f'{self.output_config.configName}'
        if res.output_explanation:
            res.output_result = f'{self.output_config.secondConfigName} does not forbid connections specified ' \
                                         f'in {self.output_config.configName}'
        return res
```

and simplify `execute_and_write_output_in_required_format` as follows:

```
    def execute_and_write_output_in_required_format(self, cmd_line_flag=False):
        query_answer = self.exec(cmd_line_flag)
        explanation_header = f'Both {self.output_config.configName} and ' \
                             f'{self.output_config.secondConfigName} allow the following connection(s)'
        return query_answer.bool_result,\
            self.write_query_output(query_answer, explanation_header, [explanation_header], int(query_answer.bool_result)), \
            query_answer.query_not_executed
```